contracts/Evolution3Serum/Evolution3Serum.sol

```diff
@@ -0,0 +1,267 @@
+    /// @notice Data structure for holding each type token information
+    /// @param maximumTokenSupply Maximum amount of a token Id allowed to mint
+    /// @param totalMintedTokenAmount Minted amount of a token Id
+    /// @param tokenId token Id value
+    struct TokenInfo {
+        uint256 maximumTokenSupply;
+        uint256 totalMintedTokenAmount;
+        uint256 tokenId;
+    }
+
+    /// @notice List of address that can mint the token
+    /// @dev Use this mapping to set permission on who can mint the token
+    /// @custom:key A valid ethereum address
+    /// @custom:value Set permission in boolean. 'true' means allowed
+    mapping(address => bool) public minterList;
+
+    /// @notice List of address that can burn the token
+    /// @dev Use this mapping to set permission on who can burn the token
+    /// @custom:key A valid ethereum address
+    /// @custom:value Set permission in boolean. 'true' means allowed
+    mapping(address => bool) public burnerList;
+
+    /// @notice List of ERC1155 token types of this smart contract
+    /// @dev Use this mapping to store token information
+    /// @custom:key tokenId
+    /// @custom:value Token information with TokenInfo structure
+    mapping(uint256 => TokenInfo) public tokenCollections;
+
+    /// @notice Total of token ID
+    uint256 public totalToken = 0;
+
+    /// @notice Base URL to store off-chain metadata information
+    /// @dev This variable could be used to store URL for the token metadata
+    string public baseURI = "";
+
+    /// @notice Check whether address is valid
+    /// @param _address Any valid ethereum address
+    modifier validAddress(address _address) {
+        if (_address == address(0)) {
+            revert InvalidAddress();
+        }
+        _;
+    }
+
+    /// @notice Check whether token ID exists
+    /// @param _tokenId token ID
+    modifier validTokenId(uint256 _tokenId) {
+        if (_tokenId < 1 || _tokenId > totalToken) {
+            revert InvalidTokenId();
+        }
+        _;
+    }
+
+    /// @notice Check whether amount not zero
+    /// @param _amount token amount
+    modifier validAmount(uint256 _amount) {
+        if (_amount == 0) {
+            revert ValueCanNotBeZero();
+        }
+        _;
+    }
+
+    /// @notice There's a mint transaction happen
+    /// @dev Emit event when calling mint function
+    /// @param minter Address who calls the function
+    /// @param receiver Address who receives the token
+    /// @param tokenId Item token ID
+    /// @param amount Amount of item that is minted
+    event E3SerunMint(address minter, address receiver, uint256 tokenId, uint256 amount);
+
+    /// @notice There's a burn transaction happen
+    /// @dev Emit event when calling burn function
+    /// @param walletAddress Address of the token holder
+    /// @param tokenId Token to burnt
+    /// @param amount Amount of token to burn
+    event E3SeumBurn(address walletAddress, uint256 tokenId, uint256 amount);
```

Comment:
I think there is a mistype, "E3SeumBurn" suppose to be "E3Se**r**umBurn"

Reply:
fixed